testing/mjolnir/src/mjolnir_lib/generators/fragment_generator.rs

```diff
@@ -34,30 +41,39 @@ pub struct FragmentGenerator<'a, S: SyncNode + Send> {
     stake_pools_count: usize,
     vote_plans_for_tally_count: usize,
     vote_plans_for_casting_count: usize,
+    update_proposals_for_casting_count: usize,
+    fragment_options_count: usize,
 }
 
 impl<'a, S: SyncNode + Send> FragmentGenerator<'a, S> {
     #[allow(clippy::too_many_arguments)]
     pub fn new(
```

Comment:
The code is ok, i just wanted to say that if you want and have time, maybe it will be better to introduce builder instead on this enormous `new`  method. Also you could change assert! to just Errors